### ISSUE TITLE
Come up with a fix for the usability issues

### ISSUE THREAD

**Issue by jmaassen** [research-software-directory/research-software-directory#334]
There are quite a few "quirks" in the RSD you need to know when filling in the data, such some field which must be filled, refreshing of the site, caching, etc. This makes it difficult to enter data for a non-expert user, as there is often little or no feedback (or no immediate feedback). 

We should come up with a way to make all of this more intuitive.

**Comment by jmaassen** [comment]
Also related to #333

**Comment by jspaaks** [comment]
I agree with the general sentiment of the issue, but it would help development/decision making if this was more specific. At least we'd need to have enough information to know when to consider this issue resolved.

**Comment by sverhoeven** [comment]
A more specific issue #638 was spawned from this issue and was solved in PR #639. So now after adding for example a new person you can go to a software item and see that new person without having to reload the page manually.

**Comment by jiskattema** [comment]
Hi, i want to add to the usability issues. I'm manually adding projects that are just starting.
For hardly any of the *required* fields, I have the sensible information (URLs, dois, .. ).
And in some cases the required fields do not make sense.
Even for making drafts, i have to guess (copy from the error message) something to fool the regexp.

**Comment by jspaaks** [comment]
Hi Jisk,

>  For hardly any of the required fields, I have the sensible information (URLs, dois, .. ).

Not sure what can be done about that in terms of the RSD though. Perhaps I'm missing something?

>  And in some cases the required fields do not make sense.

That's possible, the last time the schema was updated was long before Programme Managers were a thing. I don't have any available resources at the moment to make any changes unfortunately. But I guess we can fill in the things that do make sense and call it a day?

>  Even for making drafts, i have to guess (copy from the error message) something to fool the regexp.

Can you give an example?

**Comment by jiskattema** [comment]
Hi jurriaan! thanks for replying. The external URL/links (https://doi.org/., github repositories, ... ) and things like subtitle all validate their input while typing, and seem to **prevent me from using 'save' to save my draft**.
I understand that publishing would require some fields to be filled in properly (ie. they validate against a regexp or so..), but for saving a draft this is counter productive.
Also, what if i have a gitlab instead of github URL? At the moment the page refuses to accept that.

**Comment by jspaaks** [comment]
Agree, that validation-while-typing thingy is an absolute nuisance. When I have to type longer things, I copy pste them from another place. Not ideal, but saves me headaches. Additionally, for some non required fields you can trick the validator by typing a character and removing it again.

The draft saving has been raised before, I agree this is annoying.

GitLab repositories are not recognized because our policy says: use GitHub. I realize this isn't always realistic, for example when projects bring existing code bases. Maybe we should start suggesting our engineers to set up a fork and automated sync with GitHub? Since other systems like metrics, graphs, tractability etc, depend on GitHub.

Hope this helps, I can help you with other things if need be. Either via this or Teams. Just let me know.
-Jurriaan

**Comment by jiskattema** [comment]
thanks for the reply.

> GitLab repositories are not recognized because our policy says: use GitHub. I realize this isn't always realistic, for
> example when projects bring existing code bases. Maybe we should start suggesting our engineers to set up a fork 
> and automated sync with GitHub? Since other systems like metrics, graphs, tractability etc, depend on GitHub.

i understand the RSD is written from a software/data harvesting perspective, but many of our projects dont fit that template. Existing (sometimes non-github) repos are often even better than our own: the project hopefully continues without us at a later date. And having lots of abandoned (because our project ended) forks around is also not nice. At the minimum we then should start with a 'this repo is archived, see upstream for further development' policy etc.

**Comment by jspaaks** [comment]
>  And having lots of abandoned (because our project ended) forks around is also not nice

Having a dead fork is not a problem for me. E.g. I don't expect all these forks to start sending us PRs anytime soon https://github.com/NLeSC/mcfly/network/members

Looks like GitLab can be configured to maintain a mirror on GitHub: https://docs.gitlab.com/ee/user/project/repository/mirror/push.html.
